compiler/resolution/cullOverReferences.cpp

```diff
@@ -493,19 +493,26 @@ static void maybeIssueRefMaybeConstWarning(ArgSymbol* arg) {
 
     const char* argName = nullptr;
     char argBuffer[64];
-    if (!arg->hasFlag(FLAG_EXPANDED_VARARGS)) {
-      argName = arg->name;
-    } else {
+    if (isTaskIntent && arg->hasFlag(FLAG_FIELD_ACCESSOR)) {
+      sprintf(argBuffer, "this");
+      argName = argBuffer;
+    } else if (arg->hasFlag(FLAG_EXPANDED_VARARGS)) {
       int varArgNum;
       int ret = sscanf(arg->name, "_e%d_%63s", &varArgNum, argBuffer);
       CHPL_ASSERT(ret == 2);
       argName = argBuffer;
+    } else {
+      argName = arg->name;
     }
 
-    USR_WARN(arg,
-              "inferring a default intent to be 'ref' is deprecated "
-              "- please use an explicit intent for the argument '%s'",
-              argName);
+    const char* intentName = isTaskIntent ? "add an explicit task variable for" : (isArgThis ? "use an explicit this intent for the method" : "use an explicit intent for the argument");
```

Comment:
"add an explicit task variable": "variable" --> "intent"

Also I suggest we add "'ref'" in all these three suggestions.

It would also be nicer to split this into two/three lines to fit within 79 columns each.

Be sure to update .good file(s).

Reply:
Ex. `please add an explicit 'ref' task intent for XYZ`, if this makes sense.